### DISCUSSION TITLE
Official way to hook into fast refresh for external content

### DISCUSSION THREAD

**Discussion by Svish** [vercel/next.js#23073] [Ideas]
### Describe the feature you'd like to request

When developing it's common to fetch data from "outside" of Next.js. Could be files on disk, a database, a separate API, etc. Currently, the Next.js Fast Refresh feature of course will not pick up changes in these sources, but I really wish there was an official way to do that as it would speed up local development a lot. Fast Refresh on pages, components, etc., is super awesome, but often a big part of developing is editing content outside of Next.js. And it would be SUPER awesome if Fast Refresh was usable in that context too.

### Describe the solution you'd like

An official way to make the Fast Refresh also do a refresh based on external sources. Should also come with a few easy to setup "plugins" for common sources, like watching a file-directory, receiving a POST event (e.g. from a CMS), etc.

Basically, something like [next-remote-watch](https://github.com/hashicorp/next-remote-watch), except officially supported and built-in to `next dev`.

### Describe alternatives you've considered

I've come across [next-remote-watch](https://github.com/hashicorp/next-remote-watch), which I haven't tried yet. It looks promising, but as it says:

> This package utilizes undocumented APIs from next.js that are not subject to semantic versioning. This means that any version bump to next.js, major, minor, or patch, could cause it to break without warning.

It also tells you to replace the `next dev` command, which is of course needed, but wish this was a feature built-in to `next dev` instead.
